test/message/test_runner_describe_it.js

```diff
@@ -339,3 +339,47 @@ describe('timeouts', () => {
     setTimeout(done, 10);
   });
 });
+
+describe('Concurrency option (boolean) = true ', { concurrency: true }, () => {
+  const suiteStartDateTime = new Date().getTime();
+  const extraBufferTime = 10;
+  it('should be over by 3 seconds from suite start', async () => {
+    const duration = 3000;
+    const expectedEndDateTime = suiteStartDateTime + duration + extraBufferTime;
+    await new Promise((resolve, reject) => {
+      setTimeout(resolve, duration);
+    });
+    assert.strictEqual(expectedEndDateTime > new Date().getTime(), true);
+  });
+  it('should be over by 1 second from suite start', async () => {
+    const duration = 1000;
+    const expectedEndDateTime = suiteStartDateTime + duration + extraBufferTime;
+    await new Promise((resolve, reject) => {
+      setTimeout(resolve, duration);
+    });
+    assert.strictEqual(expectedEndDateTime > new Date().getTime(), true);
+  });
+});
+
+describe('Concurrency option (boolean) = false ', { concurrency: false }, () => {
+  const suiteStartDateTime = new Date().getTime();
+  const extraBufferTime = 10;
+  it('should be over by 3 seconds from suite start', async () => {
+    const duration = 3000;
+    const expectedEndDateTime = suiteStartDateTime + duration + extraBufferTime;
+    await new Promise((resolve, reject) => {
+      setTimeout(resolve, duration);
+    });
+    assert.strictEqual(expectedEndDateTime > new Date().getTime(), true);
+  });
+  it('should be over by 4 seconds from suite start', async () => {
+    const duration = 1000;
+    const prevTestDuration = 3000;
+    const expectedEndDateTime =
+      suiteStartDateTime + duration + extraBufferTime + prevTestDuration;
+    await new Promise((resolve, reject) => {
+      setTimeout(resolve, duration);
+    });
+    assert.strictEqual(expectedEndDateTime > new Date().getTime(), true);
+  });
+});
```

Comment:
We could simplify the test with `common.mustCall` and `common.mustNotCall`:
```suggestion
describe('Concurrency option (boolean) = true ', { concurrency: true }, () => {
  it('should never end', () => new Promise(() => {}));
  it('should start anyway', common.mustCall());
});
describe('Concurrency option (boolean) = false ', { concurrency: false }, () => {
  it('should never end', () => new Promise(() => {}));
  it('should never start', common.mustNotCall());
});
```

Reply:
@aduh95 I tried this out, but the parent test kills the long-running sub-test which would lead the entire test to fail.